.github/workflows/pull_request.yml

```diff
@@ -93,18 +95,27 @@ jobs:
             sudo dpkg -i wkhtmltox_0.12.6-1.bionic_amd64.deb
       - name: Write app version
         run: printf "${{ github.event.pull_request.head.sha }}" > .application-version
-      - name: Install pipenv
-        run: pip install pipenv==2023.8.22
-      - name: Install virtual environment
-        run: pipenv install --dev
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.8.2
+          virtualenvs-create: true
+      - uses: actions/setup-python@v5
+        id: cache-poetry
```

Comment:
id is now redundant, hence can be removed.

Reply:
Removed